python/lib/communication/dmod/communication/client.py

```diff
@@ -117,133 +161,266 @@ async def async_recv(self) -> str:
 
     @property
     @abstractmethod
-    def client_ssl_context(self) -> ssl.SSLContext:
+    def endpoint_uri(self) -> str:
```

Comment:
Does this _need_ to be public? Just trying to reduce the public interface to only what is required.

Reply:
I'll change this also for now, but it (or something close to it) may have to come back for #417.  My reasoning there may have actually been circular, though, so maybe it is superfluous.